### PR TITLE
Fixes #7868: Add calls to local API in rudder server commands

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/7868

### CODE REVIEW

**Comment by amousset** [comment]
PR updated

File: share/lib/api_call.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
```
Comment (peckpeck) [inline share/lib/api_call.sh]: This file is not meant to be used directly, please remove this line

File: share/commands/server-reload-groups [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    i|I|v|d)
+      DISPLAY_COMMAND=true
+      ;;
+    c)
+      clear_colors
+      ;;
+  esac
+done
+
+simple_api_call "https://localhost/rudder/api/dyngroup/reload" "reload dynamic groups" "${DISPLAY_COMMAND}"
```
Comment (peckpeck) [inline share/commands/server-reload-groups]: missing newline

File: share/commands/server-reload-techniques [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    i|I|v|d)
+      DISPLAY_COMMAND=true
+      ;;
+    c)
+      clear_colors
+      ;;
+  esac
+done
+
+simple_api_call "http://localhost/rudder/api/techniqueLibrary/reload" "reload techniques from files" "${DISPLAY_COMMAND}"
```
Comment (peckpeck) [inline share/commands/server-reload-techniques]: missing newline

File: share/lib/api_call.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+# This file provide helpers to make API call from Rudder commands
+
+. "${BASEDIR}/../lib/common.sh"
+
+CURL="/usr/bin/curl"
+
+# This functions tests if the API call returns "OK"
+function simple_api_call() {
```
Comment (peckpeck) [inline share/lib/api_call.sh]: just use the xxx() {} syntax, without "function"

File: share/lib/api_call.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# This file provide helpers to make API call from Rudder commands
+
+. "${BASEDIR}/../lib/common.sh"
+
+CURL="/usr/bin/curl"
```
Comment (peckpeck) [inline share/lib/api_call.sh]: You can do better for the command, for example in rudder-setup: 
# get a remote url content using the first available method
get() {
  WGET="wget -q -O"
  CURL="curl -s -o"
  if type curl >/dev/null 2>/dev/null
  then
    ${CURL} "$@"
  else
    ${WGET} "$@"
  fi
}

File: share/lib/api_call.sh [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  action="$2"
+  display_command="$3"
+  curl_command="${CURL} -sS -k -L \"${url}\""
+  if $display_command;
+  then
+    printf "${WHITE}${curl_command}${NORMAL}\n"
+  fi
+  result=`eval ${curl_command}`
+  code=$?
+  if [ $code -eq 0 ] && [ "$result"="OK" ]
```
Comment (peckpeck) [inline share/lib/api_call.sh]: missing spaces around =

Comment (peckpeck) [inline share/lib/api_call.sh]: Prefer the form ${} for variables

File: share/lib/api_call.sh [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  then
+    printf "${GREEN}ok${NORMAL}: ${action}.\n"
+  else
+    printf "${RED}error${NORMAL}: Could not ${action}\n"
+    echo ${result}
```
Comment (peckpeck) [inline share/lib/api_call.sh]: Use "" for arguments

**Comment by peckpeck** [comment]
ping @amousset

**Comment by amousset** [comment]
PR updated

**Comment by amousset** [comment]
PR updated

**Comment by amousset** [comment]
@peckpeck PR updated.

**Comment by peckpeck** [comment]
Good, merging

**Comment by jooooooon** [comment]
Wow! These commands are invaluable - what a time saver for upgrades, etc.

Once it has been tested a bit and proven to work OK, I think we should backport this to 3.1, like we did the rudder agent * commands. I will open a ticket to discuss this.

**Pull request merged**